src/libImaging/Draw.c

```diff
@@ -1347,6 +1347,22 @@ pie_init(clip_ellipse_state *s, int32_t a, int32_t b, int32_t w, float al, float
     s->root->l = lc;
     s->root->r = rc;
     s->root->type = ar - al < 180 ? CT_AND : CT_OR;
+
+    // add one more semiplane to avoid spikes
+    if (ar - al < 90) {
+        clip_node *old_root = s->root;
+        clip_node *spike_clipper = s->nodes + s->node_count++;
+        s->root = s->nodes + s->node_count++;
+        s->root->l = old_root;
+        s->root->r = spike_clipper;
+        s->root->type = CT_AND;
+
+        spike_clipper->l = spike_clipper->r = NULL;
+        spike_clipper->type = CT_CLIP;
+        spike_clipper->a = (xl + xr) / 2.0;
+        spike_clipper->b = (yl + yr) / 2.0;
```

Comment:
Here's my explanation of what is happening here.
This is taking the average of the two points on the outside of the circle. Going through the center of the circle as well, that gives us a line down the middle of the pieslice.
Then by swapping x and y (normally y would be with a), and multiplying one of them by -1, it gives us a perpendicular line. Because the line also goes through the center of the circle, that cuts off the spike.

Reply:
yeah, that's one way to put it
line equation Ax+By+C=0 can be also written as rn + C = 0, r(x, y), n(A, B) - i.e. n(A, B) is a normal vector for line